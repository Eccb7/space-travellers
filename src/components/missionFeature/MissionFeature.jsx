import React from 'react';
import PropTypes from 'prop-types';
import './missionFeature.css';
import { useDispatch } from 'react-redux';
import { LeaveMission, joinMission } from '../../redux/missions/missionSlice';

function MissionFeature({
  missionId, missionName, missionDesc, missionReserved,
}) {
  const dispatch = useDispatch();

  const handleJoinMission = (id) => {
    dispatch(joinMission(id));
  };
  const handleLeaveMission = (id) => {
    dispatch(LeaveMission(id));
  };

  return (
    <tr>
      <td className="app_mission-title">{missionName}</td>
      <td>{missionDesc}</td>
      {!missionReserved
        ? (<td><span>NOT A MEMBER</span></td>)
        : (<td><span>Active Member</span></td>)}
      {!missionReserved
        ? (<td><button type="button" title="Join Mission" onClick={() => handleJoinMission(missionId)}>Join Mission</button></td>)
        : (<td><button type="button" title="Leave Mission" onClick={() => handleLeaveMission(missionId)}>Leave Mission</button></td>)}
    </tr>
  );
}

MissionFeature.propTypes = {
  missionId: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  missionDesc: PropTypes.string.isRequired,
  missionReserved: PropTypes.bool.isRequired,
};

export default MissionFeature;
