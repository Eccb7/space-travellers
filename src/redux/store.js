import { configureStore } from '@reduxjs/toolkit';
import { createLogger } from 'redux-logger';
import missionReducer from './missions/missionSlice';

const logger = createLogger();
const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
