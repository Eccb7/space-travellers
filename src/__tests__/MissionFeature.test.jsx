import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MissionFeature from '../components/missionFeature/MissionFeature';

test('should render MissionFeature correctly when missionReserved is false', () => {
  const props = {
    missionId: '1',
    missionName: 'Mission 1',
    missionDesc: 'Description for Mission 1',
    missionReserved: false,
  };
  const { container } = render(
    <Provider store={store}>
      <MissionFeature
        missionId={props.missionId}
        missionName={props.missionName}
        missionDesc={props.missionDesc}
        missionReserved={props.missionReserved}
      />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
