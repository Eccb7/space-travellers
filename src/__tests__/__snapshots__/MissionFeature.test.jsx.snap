// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionFeature should render MissionFeature correctly when missionReserved is false 1`] = `
<div>
  <tr>
    <td
      class="app_mission-title"
    >
      Mission 1
    </td>
    <td>
      Description for Mission 1
    </td>
    <td>
      <span
        class="app_mission-status app_mission-status_nomember"
      >
        NOT A MEMBER
      </span>
    </td>
    <td>
      <button
        class="app_mission-button app_mission-button_join"
        title="Join Mission"
        type="button"
      >
        Join Mission
      </button>
    </td>
  </tr>
</div>
`;

exports[`should render MissionFeature correctly when missionReserved is false 1`] = `
<div>
  <tr>
    <td
      class="app_mission-title"
    >
      Mission 1
    </td>
    <td>
      Description for Mission 1
    </td>
    <td>
      <span
        class="app_mission-status app_mission-status_nomember"
      >
        NOT A MEMBER
      </span>
    </td>
    <td>
      <button
        class="app_mission-button app_mission-button_join"
        title="Join Mission"
        type="button"
      >
        Join Mission
      </button>
    </td>
  </tr>
</div>
`;
